src/plugins/vis_augmenter/public/view_events_flyout/components/utils.tsx

```diff
@@ -43,6 +43,45 @@ function getValueAxisPositions(embeddable: VisualizeEmbeddable): { left: boolean
   };
 }
 
+/**
+ * Fetching the base vis to show in the flyout, based on the saved object ID. Add constraints
+ * such that it is static and won't auto-refresh within the flyout.
+ * @param savedObjectId the saved object id of the base vis
+ */
+export async function fetchVisEmbeddable(savedObjectId: string): Promise<VisualizeEmbeddable> {
+  const embeddableVisFactory = getEmbeddable().getEmbeddableFactory('visualization');
+  const contextInput = {
+    filters: getQueryService().filterManager.getFilters(),
+    query: getQueryService().queryString.getQuery(),
+    timeRange: getQueryService().timefilter.timefilter.getTime(),
+  };
+
+  const embeddable = (await embeddableVisFactory?.createFromSavedObject(savedObjectId, {
+    ...contextInput,
+    visAugmenterConfig: {
+      inFlyout: true,
+      flyoutContext: VisFlyoutContext.BASE_VIS,
+    },
+  } as VisualizeInput)) as VisualizeEmbeddable | ErrorEmbeddable;
+
+  if (embeddable instanceof ErrorEmbeddable) {
+    throw getErrorMessage(embeddable);
+  }
+
+  embeddable.updateInput({
+    // @ts-ignore
```

Comment:
nit - all `@ts-ignore` statements should have explanatory comments. I'm also curious why it's necessary here - is it because you're just passing a partial input?

Reply:
This was existing code moved here. The reason for the ignore here is that the embeddable is within a dashboard container that controls a `refreshConfig` field - this isn't included as part of the default `embeddableInput` that it resolves to, so IDEs complain about it

Reply:
can you either add that to a comment or open an issue?